drivers/hyperv/hyperv.go

```diff
@@ -271,39 +275,37 @@ func (d *Driver) Create() error {
 }
 
 func (d *Driver) chooseVirtualSwitch() (string, error) {
-	if d.VSwitch == "" {
-		// Default to the first external switche and in the process avoid DockerNAT
-		stdout, err := cmdOut("[Console]::OutputEncoding = [Text.Encoding]::UTF8; (Hyper-V\\Get-VMSwitch -SwitchType External).Name")
-		if err != nil {
-			return "", err
-		}
-
-		switches := parseLines(stdout)
+	if strings.ToLower(d.VSwitch) == strings.ToLower(dockerNATSwitch) {
+		return "", fmt.Errorf("vswitch %s cannot be used", dockerNATSwitch)
+	}
 
-		if len(switches) < 1 {
-			return "", fmt.Errorf("no External vswitch found. A valid vswitch must be available for this command to run. Check https://docs.docker.com/machine/drivers/hyper-v/")
-		}
+	stdout, err := cmdOut("[Console]::OutputEncoding = [Text.Encoding]::UTF8; ConvertTo-Json @(Hyper-V\\Get-VMSwitch | Select Id, Name, SwitchType)")
+	if err != nil {
+		return "", err
+	}
 
-		return switches[0], nil
+	type Switch struct {
+		ID         string
+		Name       string
+		SwitchType int
 	}
 
-	stdout, err := cmdOut("[Console]::OutputEncoding = [Text.Encoding]::UTF8; (Hyper-V\\Get-VMSwitch).Name")
+	var switches []Switch
+	err = json.Unmarshal([]byte(strings.NewReplacer("\r", "").Replace(stdout)), &switches)
 	if err != nil {
 		return "", err
 	}
 
-	switches := parseLines(stdout)
-
 	found := false
-	for _, name := range switches {
-		if name == d.VSwitch {
+	for _, vSwitch := range switches {
+		if vSwitch.Name == d.VSwitch {
 			found = true
 			break
 		}
 	}
 
 	if !found {
-		return "", fmt.Errorf("vswitch %q not found", d.VSwitch)
+		return defaultSwitch, nil
```

Comment:
Pretty sure this default switch is only available after a certain version of Windows 10. What if the user is working on a lower OS and this default switch doesn't exist?

Reply:
The latest commit should tell the user about the suggestion when the `External` vSwitch and `Default Switch` does not exist